### PR TITLE
Newtab - PR for notes

### PR DESCRIPTION
**Pull request by winterstein**
Not ready for merge yet - but this early PR is a handy way to pass comments around

### CODE REVIEW

File: src/js/components/NewTabMainDiv.jsx [hunk 1]
```diff
@@ -50,66 +59,93 @@ let bg = randomPick([
 
 const WebtopPage = () => {
 
+	const searchRef = useRef(null);
```
Comment (winterstein) [inline src/js/components/NewTabMainDiv.jsx:62]: Please add a doc comment -- Why is useRef() wanted here?

File: src/js/components/NewTabMainDiv.jsx [hunk 2]
```diff
@@ -50,66 +59,93 @@ let bg = randomPick([
...
-	if ( ! Login.isLoggedIn()) {		
-		return <LoginLink />;
-	}
-	let user = Login.getUser();
-	return <div>
-		{user.name || user.id}
-		<small><LogoutLink /></small>
-	</div>;
+const Search = () => {
+	return (<>
```
Comment (winterstein) [inline src/js/components/NewTabMainDiv.jsx:115]: Hm - We should tell the user that this search is via Ecosia, and will raise money for planting trees.

File: src/js/components/Ticker.jsx [hunk 3]
```diff
@@ -66,22 +66,16 @@ const Ticker = ({value, amount, rate, tickTime=1000, currencySymbol = '', pretty
...
+	let dispArr = disp.split("");
 	return (
-		<div className="position-relative d-inline-block" style={{padding: "0 " + (centerText ? "0.1rem" : "0")}}>
-			<span className="invisible text-center" style={{width: centerText ? "100%" : "auto"}}>{currencySymbol + totalVal}</span>
-			<span className="position-absolute text-center" style={{right: 0, width: centerText ? "100%" : "auto"}} ref={ref}>{currencySymbol + disp}</span>
+		<div className="position-relative flex-row justify-content-center align-items-center" style={{padding: "0 " + (centerText ? "0.1rem" : "0")}}>
+			{/*<span className="invisible text-center" style={{width: centerText ? "100%" : "auto"}}>{currencySymbol + totalVal}</span>
+			<span className="position-absolute text-center" style={{right: 0, width: centerText ? "100%" : "auto"}} ref={ref}>{currencySymbol + disp}</span>*/}
+			{dispArr.map(digit => <div style={{width:unitWidth || "1rem", textAlign:"center", margin:"unset"}}>
+				{digit}
```
Comment (winterstein) [inline src/js/components/Ticker.jsx:77]: map() outputs are missing the `key` prop

File: src/style/NewTab.less [hunk 4]
```diff
@@ -0,0 +1,87 @@
+.tab-user {
+    font-family: "Tajawal", sans-serif;
```
Comment (winterstein) [inline src/style/NewTab.less:2]: Please avoid local font css

File: src/style/NewTab.less [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    font-weight: 600;
+    .tab-login {
+        overflow: hidden;
+        text-transform: uppercase;
+        .tab-login-content {
+            transform: translateX(-101%);
```
Comment (winterstein) [inline src/style/NewTab.less:8]: Please add a comment

File: src/style/NewTab.less [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            transition: 0.1s;
+        }
+    }
+    &:hover {
+        .tab-login {
+            .tab-login-content {
+                transform: translateX(0);
```
Comment (winterstein) [inline src/style/NewTab.less:15]: Please add a comment

File: src/style/NewTab.less [hunk 7]
```diff
@@ -0,0 +1,87 @@
+.tab-user {
+    font-family: "Tajawal", sans-serif;
+    font-weight: 600;
+    .tab-login {
```
Comment (winterstein) [inline src/style/NewTab.less:4]: Please add a comment on why login here is different from our normal widget. Thanks.

File: src/js/components/pages/AccountPage.jsx [hunk 8]
```diff
@@ -19,6 +20,34 @@ import SubscriptionBox from '../cards/SubscriptionBox';
...
+const tabs = {
+	account: {
+		content: <Account/>,
+		name: "My Account"
+	},
+	tabsForGood: {
+		content: <TabsForGoodSettings/>,
+		name: "Tabs for Good"
+	},
+};
```
Comment (winterstein) [inline src/js/components/pages/AccountPage.jsx:50]: This kind of local ad-hoc data+code structure makes for confusing code. 
Instead, try a list of tab names for the sidebar, and then a `switch(name)` when you need to make the jsx.

File: src/style/AccountPage.less [hunk 9]
```diff
@@ -61,6 +61,25 @@ A .TubeLine div contains seberal sections with TubeLine-img and TubeLine-line
 
 .AccountPage {
 
+	.account-sidebar {
+		h5 {
+			text-transform: uppercase;
+			font-weight: 600;
+			font-family: 'Montserrat', sans-serif;
+		}
```
Comment (winterstein) [inline src/style/AccountPage.less:69]: @dodo721 
Avoid specific font overrides unless they are needed - in which case put a comment in to explain why.
Put the comment in at the same time as writing the rule.

Section specific rules make for bloated and fragile css -- and inconsistent styling. They are often the right thing - But the reason must be documented.

File: src/style/AccountPage.less [hunk 10]
```diff
@@ -73,6 +92,9 @@ A .TubeLine div contains seberal sections with TubeLine-img and TubeLine-line
 			font-size: 1.6rem;
```
Comment (winterstein) [inline src/style/AccountPage.less:92]: Avoid ad-hoc font settings, e.g. size

File: src/style/CampaignPage.less [hunk 11]
```diff
@@ -148,7 +141,7 @@ body {
 					font-size: 1.6rem;
 					color: @gl-orange;
 					font-weight: 700;
-					@media (max-width:1400px) {
+					@media @lg-max {
 						font-size: 1.4rem;
```
Comment (winterstein) [inline src/style/CampaignPage.less:145]: Avoid ad-hoc font settings, e.g. size

File: src/style/NewTab.less [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+        }
+    }
+}
+
+.tab-footer {
+    position: fixed;
+    bottom: 0;
+    left: 50%;
+    transform: translate(-50%, 90%);
+    transition: 0.2s;
```
Comment (winterstein) [inline src/style/NewTab.less:27]: Please comment

File: src/style/NewTab.less [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+.tutorial-highlight {
+    // Border + shadow = glow
+    border: 3px solid @gl-light-red;
+    box-shadow: 0 0 10px @gl-light-red;
+}
+
+.onboarding {
+    h1 {
+        font-size: 3.5rem;
```
Comment (winterstein) [inline src/style/NewTab.less:88]: avoid ad-hoc font settings

File: src/style/NewTab.less [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+        text-shadow: 3px 3px 6px #00000077;
+    }
+}
+
+#webtop {
+    h1,h2,h3,h4,h5 {
+        color: white;
+        font-weight: 600;
+        font-family: 'Montserrat', sans-serif;
+    }
```
Comment (winterstein) [inline src/style/NewTab.less:102]: avoid ad-hoc font settings

File: src/style/NewTab.less [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+
+    .logo {
+        h4 {
+            text-transform: uppercase;
+            font-weight: 800;
+        }
```
Comment (winterstein) [inline src/style/NewTab.less:108]: avoid ad-hoc font settings

File: src/style/NewTab.less [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+.extension-btn {
+    background: @gl-light-red;
+    box-shadow: 0px 3px 6px rgba(0,0,0,0.25);
+    padding: 0.75rem 1rem;
+    text-transform: uppercase;
+    color: white !important;
+    font-weight: 600;
+    font-family: 'Montserrat', sans-serif;
```
Comment (winterstein) [inline src/style/NewTab.less:119]: avoid ad-hoc font settings

File: src/style/NewTab.less [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+
+    .logo {
+        h4 {
+            text-transform: uppercase;
+            font-weight: 800;
+        }
+    }
+}
+
+.extension-btn {
```
Comment (winterstein) [inline src/style/NewTab.less:112]: avoid ad-hoc widgets -- Use BS vanilla buttons, or document why they're not right here.

File: src/style/NewTab.less [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+    transition: 0.2s;
+
+    &:hover {
+        background: @gl-red;
+        box-shadow: 0px 3px 6px rgba(0,0,0,0.5);
+    }
+}
+
+.tab-login-widget {
+    h1,h2,h3, h4 {
```
Comment (winterstein) [inline src/style/NewTab.less:130]: Why is login different here?

File: src/style/TabsForGoodSettings.less [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+        &.do3d {
+            transform-style: preserve-3d;
+            * {
+                transition: all 0.5s ease;
+            }
+        }
+        a {
+            color: @gl-light-red;
+            font-weight: 600;
```
Comment (winterstein) [inline src/style/TabsForGoodSettings.less:25]: avoid ad-hoc font settings

File: src/style/main.less [hunk 20]
```diff
@@ -149,7 +149,7 @@ a.btn {
 	}
 
 	.btn {
-		font-size: 100%; // Override Bootstrap's explicit px sizing for contextual sizing
+		font-size: 1rem;
 	}
```
Comment (winterstein) [inline src/style/main.less:153]: why?

File: src/js/components/NewTabMainDiv.jsx (hunk 2 above)
Comment (dodo721) [reply src/js/components/NewTabMainDiv.jsx]: Added placeholder text

File: src/js/components/Ticker.jsx (hunk 3 above)
Comment (dodo721) [reply src/js/components/Ticker.jsx:88]: Fixed

File: src/style/NewTab.less (hunk 5 above)
Comment (dodo721) [reply src/style/NewTab.less:8]: Outdated, removed

File: src/style/NewTab.less (hunk 6 above)
Comment (dodo721) [reply src/style/NewTab.less:15]: Outdated, removed

File: src/style/NewTab.less (hunk 7 above)
Comment (dodo721) [reply src/style/NewTab.less:4]: Outdated, removed

File: src/js/components/pages/AccountPage.jsx (hunk 8 above)
Comment (dodo721) [reply src/js/components/pages/AccountPage.jsx:40]: Refactored

**Pull request merged**